websites/L/LTTStore.com/presence.ts

```diff
@@ -0,0 +1,54 @@
+const presence = new Presence({
+	clientId: "1258619078201839646",
+});
+
+const enum Assets {
+	Logo = "https://upload.wikimedia.org/wikipedia/commons/thumb/7/77/2018_Linus_Tech_Tips_logo.svg/512px-2018_Linus_Tech_Tips_logo.svg.png",
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: Assets.Logo,
+	};
+	if (document.location.pathname === "/" || document.location.pathname === "") {
+		presenceData.details = "Browsing";
+		presenceData.state = "Home";
+	} else if (document.location.pathname.includes("/products/")) {
+		presenceData.largeImageKey =
+			document.querySelector<HTMLMetaElement>('[property="og:image"]')
+				?.content ?? Assets.Logo;
+
+		presenceData.smallImageKey = Assets.Logo;
+		presenceData.smallImageText = "LTTStore.com";
+		presenceData.details = "Viewing Product";
+		presenceData.state = (
+			document.querySelector("div.product__title h1") as HTMLElement
```

Comment:
Use generics rather than casting. In this case, a specific type might not be needed.

Reply:
Im not entirely sure as to what you mean. I put in the commit what I believe is what you mean but I'll leave it up to review